### PR TITLE
add C# rules for PRs and issues

### PR DESCRIPTION
**Pull request by BillWagner**
Apply area and technology labels to issues and PRs in the C# guide.

### CODE REVIEW

File: .ghal.rules.json [hunk 1]
```diff
@@ -53,6 +53,36 @@
...
+            "labels-add": ":card_file_box: Technology - C# Get Started"
+          },
+          "(?i)csharp-language-reference$": {
+            "labels-add": ":card_file_box: Technology - C# Language Reference"
+          },
+          "(?i)csharp-linq$": {
+            "labels-add": ":card_file_box: Technology - LINQ"
+          },
+          "(?i)csharp-null-safety$": {
+            "labels-add": ":card_file_box: Technology - C# Null safety"
```
Comment (pkulikov) [inline .ghal.rules.json:78]:
```suggestion
            "labels-add": ":card_file_box: Technology - C# null safety"
```

Comment (BillWagner) [reply .ghal.rules.json:78]: Thanks @pkulikov  eagle :eyes: there.  
I'd proofread this, and missed that one.

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

File: .ghal.rules.json [hunk 2]
```diff
@@ -53,6 +53,36 @@
...
+            "labels-add": ":card_file_box: Technology - C# Get Started"
+          },
+          "(?i)csharp-language-reference$": {
+            "labels-add": ":card_file_box: Technology - C# Language Reference"
+          },
+          "(?i)csharp-linq$": {
+            "labels-add": ":card_file_box: Technology - LINQ"
+          },
+          "(?i)csharp-null-safety$": {
+            "labels-add": ":card_file_box: Technology - C# null safety"
```
Comment (gewarren) [inline .ghal.rules.json:78]:
```suggestion
            "labels-add": ":card_file_box: Technology - C# Null Safety"
```

File: .ghal.rules.json [hunk 3]
```diff
@@ -205,6 +235,117 @@
...
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/types.*": {
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/statements-expressions-operators.*": {
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/interop.*": {
+                "labels-add": ":card_file_box: Technology -  C# Advanced concepts"
```
Comment (gewarren) [inline .ghal.rules.json:269]:
```suggestion
                "labels-add": ":card_file_box: Technology -  C# Advanced Concepts"
```

File: .ghal.rules.json [hunk 4]
```diff
@@ -205,6 +235,117 @@
...
+              },
+              "(?i).*docs\/csharp\/programming-guide\/unsafe-code-pointers.*": {
+                "labels-add": ":card_file_box: Technology -  C# Advanced concepts"
```
Comment (gewarren) [inline .ghal.rules.json:272]:
```suggestion
                "labels-add": ":card_file_box: Technology -  C# Advanced Concepts"
```

File: .ghal.rules.json [hunk 5]
```diff
@@ -205,6 +235,117 @@
...
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/nullable-types.*": {
+                "labels-add": ":card_file_box: Technology -  C# Null Safety"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/arrays.*": {
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/concepts\/covariance-contravariance.*": {
+                "labels-add": ":card_file_box: Technology -  C# Advanced concepts"
```
Comment (gewarren) [inline .ghal.rules.json:287]:
```suggestion
                "labels-add": ":card_file_box: Technology -  C# Advanced Concepts"
```

File: .ghal.rules.json [hunk 6]
```diff
@@ -205,6 +235,117 @@
...
+              },
+              "(?i).*docs\/csharp\/programming-guide\/concepts\/serialization.*": {
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/programming-guide\/concepts\/expression-trees.*": {
+                "labels-add": ":card_file_box: Technology -  C# Advanced concepts"
```
Comment (gewarren) [inline .ghal.rules.json:293]:
```suggestion
                "labels-add": ":card_file_box: Technology -  C# Advanced Concepts"
```

File: .ghal.rules.json [hunk 7]
```diff
@@ -205,6 +235,117 @@
...
+                "labels-add": ":card_file_box: Technology -  C# Get Started"
+              },
+              "(?i).*docs\/csharp\/tutorial\/intro-to-csharp.*": {
+                "labels-add": ":card_file_box: Technology -  C# Get Started"
+              },
+              "(?i).*docs\/csharp\/tutorials.*": {
+                "labels-add": ":card_file_box: Technology -  C# Fundamentals"
+              },
+              "(?i).*docs\/csharp\/language-reference\/compiler-messages.*": {
+                "labels-add": ":card_file_box: Technology - C# / VB diagnostics"
```
Comment (gewarren) [inline .ghal.rules.json:338]:
```suggestion
                "labels-add": ":card_file_box: Technology - C# / VB Diagnostics"
```

**Review by gewarren — APPROVED** [review]
Not sure if capitalization matters, so I left some suggestions for consistency.

File: .ghal.rules.json (hunk 2 above)
Comment (BillWagner) [reply .ghal.rules.json:78]: I cross-checked with the label (the keys here must match the label text). The correct capitalization is "Null safety"

We can discuss changing in both places, but that's correct for now.

File: .ghal.rules.json (hunk 4 above)
Comment (BillWagner) [reply .ghal.rules.json:272]: This one is correct today, matching the label text.

File: .ghal.rules.json (hunk 6 above)
Comment (BillWagner) [reply .ghal.rules.json:293]: also matches the label text.

File: .ghal.rules.json (hunk 7 above)
Comment (BillWagner) [reply .ghal.rules.json:338]: This matches the label text as well.

**Pull request merged**